pkg/debug/spec.sh

```diff
@@ -142,6 +142,75 @@ add_pci_info() {
 __EOT__
 }
 
+get_modem_protocol() {
+    local sys_dev="$1"
+    local module="$(basename "$(readlink "${sys_dev}/device/driver/module")")"
+    case "$module" in
+        "cdc_mbim") echo "mbim"
+        ;;
+        "qmi_wwan") echo "qmi"
+        ;;
+        *) return 1
+        ;;
+    esac
+}
+
+get_modem_ifname() {
+    local sys_dev="$1"
+    ls "${sys_dev}/device/net"
+}
+
+get_modem_usbaddr() {
+    local sys_dev="$1"
+    local dev_path="$(readlink -f "${sys_dev}/device")"
+    while [ -e "$dev_path/subsystem" ]; do
+        if [ "$(basename "$(readlink "$dev_path/subsystem")")" != "usb" ]; then
+            dev_path="$(dirname "$dev_path")"
+            continue
+        fi
+        basename "$dev_path" | cut -d ":" -f 1 | tr '-' ':'
+        return
+    done
+}
+
+get_modem_pciaddr() {
+    local sys_dev="$1"
+    local dev_path="$(readlink -f "${sys_dev}/device")"
+    while [ -e "$dev_path/subsystem" ]; do
+        if [ "$(basename "$(readlink "$dev_path/subsystem")")" != "pci" ]; then
+            dev_path="$(dirname "$dev_path")"
+            continue
+        fi
+        basename "$dev_path"
+        return
+    done
+}
+
+# https://en.wikipedia.org/wiki/Hayes_command_set
+# Args: <command> <tty device>
+send_hayes_command() {
```

Comment:
I think send_AT_command() would be more intuitive to understand, but that's ok if you wanna leave as it is...